src/Repository/DoctrineInvoiceRepository.php

```diff
@@ -36,4 +36,12 @@ public function add(InvoiceInterface $invoice): void
         $this->entityManager->persist($invoice);
         $this->entityManager->flush();
     }
+
+    public function findOneByOrderNumber(string $orderNumber): InvoiceInterface
+    {
+        /** @var InvoiceInterface $invoice */
+        $invoice = $this->entityRepository->findOneBy(['orderNumber' => $orderNumber]);
+
+        return $invoice;
```

Comment:
Maybe just `return $this->entityRepository->findOneBy(['orderNumber' => $orderNumber]);`?

Reply:
Unfortunately, type hint placed in the comment is needed in order to make this code pass phpstan checks and 

```
/** @var InvoiceInterface $invoice */
return $this->entityRepository->findOneBy(['orderNumber' => $orderNumber]);
```

doesn't seem to work.

Reply:
I think we need to use https://github.com/phpstan/phpstan-webmozart-assert and `Assert::notNull($invoice)` / `Assert::isInstanceOf($invoice, InvoiceInterface::class)` between setting the variable and return.